app/actions/balancesActions.js

```diff
@@ -21,23 +21,42 @@ async function getBalances({ net, address, tokens }: Props) {
     endpoint = await api.getRPCEndpointFrom({ net }, api.neoscan)
   }
 
-  // token balances
-  const tokenBalances = await api.nep5.getTokenBalances(
-    endpoint,
-    tokens
-      .filter(token => !token.isUserGenerated)
-      .map(token => token.scriptHash),
-    address
+  const chunks = tokens
+    .filter(token => !token.isUserGenerated)
+    .reduce((accum, currVal, index) => {
+      if (!accum.length) {
+        accum.push([currVal.scriptHash])
+        return accum
+      }
+
+      if (accum[accum.length - 1].length < 4) {
+        accum[accum.length - 1].push(currVal.scriptHash)
+      } else {
+        accum.push([currVal.scriptHash])
+      }
+      return accum
+    }, [])
+
+  const promiseMap = chunks.map(chunk =>
+    api.nep5.getTokenBalances(endpoint, chunk, address)
   )
+  const results = await Promise.all(promiseMap)
+  const resultObject = results.reduce((result, currentObject) => {
+    Object.keys(currentObject).forEach(key => {
+      // eslint-disable-next-line
+      result[key] = currentObject[key]
+    })
+    return result
+  }, {})
 
-  const parsedTokenBalances = Object.keys(tokenBalances)
+  const parsedTokenBalances = Object.keys(resultObject)
```

Comment:
you could also return the format you need from the reduce method instead of calling object.keys on the data after calling reduce. its not a big deal though unless the data is large and we want to be efficient

Reply:
👍